src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -700,22 +740,32 @@ public override ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken
             Debug.Assert(_asyncActiveSemaphore != null);
             Debug.Assert(semaphoreLockTask != null);
 
-            // Employ async waiting based on the same synchronization used in BeginRead of the abstract Stream.
-            await semaphoreLockTask.ConfigureAwait(false);
+            if (!locked)
```

Comment:
Does this really help?  It's potentially increasing the size of the state machine to carry the extra parameter, and it's serving only to avoid awaiting an already completed task, all on the already slower path of having to read from the underlying stream.

Reply:
I have separated this change and benchmarked it and you were right, it was helping maybe 1% in the case where the BufferedStream buffer size and user buffer size were both small. I've reverted it